src/server/services/wfs3/qgswfs3handlers.cpp

```diff
@@ -1399,18 +1403,18 @@ void QgsWfs3CollectionsItemsHandler::handleRequest( const QgsServerApiContext &c
       if ( offset != 0 )
       {
         json prevLink = selfLink;
-        prevLink["href"] = QStringLiteral( "%1&offset=%2&limit=%3" ).arg( cleanedUrlAsString ).arg( std::max<long>( 0, limit - offset ) ).arg( limit ).toStdString();
+        prevLink["href"] = cleanedUrlAsString.toStdString() + QStringLiteral( "offset=%1&limit=%2" ).arg( std::max<long>( 0, offset - limit ) ).arg( limit ).toStdString();
         prevLink["rel"] = "prev";
-        prevLink["name"] = "Previous page";
+        prevLink["title"] = "Previous page";
         data["links"].push_back( prevLink );
       }
 
       if ( limit + offset < matchedFeaturesCount )
       {
         json nextLink = selfLink;
-        nextLink["href"] = QStringLiteral( "%1&offset=%2&limit=%3" ).arg( cleanedUrlAsString ).arg( std::min<long>( matchedFeaturesCount, limit + offset ) ).arg( limit ).toStdString();
+        nextLink["href"] = cleanedUrlAsString.toStdString() + QStringLiteral( "offset=%1&limit=%2" ).arg( std::min<long>( matchedFeaturesCount, limit + offset ) ).arg( limit ).toStdString();
         nextLink["rel"] = "next";
-        nextLink["name"] = "Next page";
+        nextLink["title"] = "Next page";
```

Comment:
You sure about this one? 
I didn't check the specs about the link attributes.

In any event, the HTML template probably needs updating if you change from "name" to "title".

Reply:
> You sure about this one?

Yep, the [spec](https://docs.opengeospatial.org/is/17-069r4/17-069r4.html#_link_relations) references [RFC8288](https://www.rfc-editor.org/rfc/rfc8288.html#section-3.5), e.g.
<img src="https://user-images.githubusercontent.com/20856381/221147455-ce2a8588-546a-432a-9614-f142b079cec9.png" width="500"/>

Also for the self/alternate links only `title` is used, here's a [live example](https://geo.kreis-viersen.de/ows/osm-daten/wfs3/collections/hoflaeden_nrw/items.geojson) of the what QGIS Server delivers currently.

> In any event, the HTML template probably needs updating if you change from "name" to "title".

Unless I'm missing something, the prev/next links are currently only used in the [getFeatures](https://github.com/qgis/QGIS/blob/5349efda7f14a9c8fb82a649981cfcf9f8b99c0f/resources/server/api/ogc/templates/wfs3/getFeatures.html)-template where the link name is hardcoded (only `href` is used):

https://github.com/qgis/QGIS/blob/5349efda7f14a9c8fb82a649981cfcf9f8b99c0f/resources/server/api/ogc/templates/wfs3/getFeatures.html#L8
https://github.com/qgis/QGIS/blob/5349efda7f14a9c8fb82a649981cfcf9f8b99c0f/resources/server/api/ogc/templates/wfs3/getFeatures.html#L14

---

btw, came across these issues while looking into implementing [full pagination](https://github.com/qgis/QGIS/blob/5349efda7f14a9c8fb82a649981cfcf9f8b99c0f/resources/server/api/ogc/templates/wfs3/getFeatures.html#L10-L12).

Reply:
> > You sure about this one?
> 
> Yep, the [spec](https://docs.opengeospatial.org/is/17-069r4/17-069r4.html#_link_relations) references [RFC8288](https://www.rfc-editor.org/rfc/rfc8288.html#section-3.5), e.g. <img alt="" width="500" src="https://user-images.githubusercontent.com/20856381/221147455-ce2a8588-546a-432a-9614-f142b079cec9.png">
> 

I was referring to the OAPIF specs, but I cannot find anything specific regarding `name` or `title` in a link so I guess the RFC8288 is the right reference in this case.


> Also for the self/alternate links only `title` is used, here's a [live example](https://geo.kreis-viersen.de/ows/osm-daten/wfs3/collections/hoflaeden_nrw/items.geojson) of the what QGIS Server delivers currently.
> 
> > In any event, the HTML template probably needs updating if you change from "name" to "title".
> 
> Unless I'm missing something, the prev/next links are currently only used in the [getFeatures](https://github.com/qgis/QGIS/blob/5349efda7f14a9c8fb82a649981cfcf9f8b99c0f/resources/server/api/ogc/templates/wfs3/getFeatures.html)-template where the link name is hardcoded (only `href` is used):
> 

No, that's ok but it needed to be checked.  I can't remember why I hardcoded it, maybe worth changing it to take whatever is in the link `title`.


> btw, came across these issues while looking into implementing [full pagination](https://github.com/qgis/QGIS/blob/5349efda7f14a9c8fb82a649981cfcf9f8b99c0f/resources/server/api/ogc/templates/wfs3/getFeatures.html#L10-L12).

Very welcome!

Reply:
> maybe worth changing it to take whatever is in the link title

👍 will be addressed with the upcoming PR